src/meta/src/stream/mod.rs

```diff
@@ -20,9 +20,54 @@ mod stream_manager;
 
 #[cfg(test)]
 mod test_fragmenter;
-
 pub use meta::*;
 pub use scheduler::*;
 pub use source_manager::*;
 pub use stream_graph::*;
 pub use stream_manager::*;
+
+/// Set vnode mapping for stateful operators.
+#[macro_export]
+macro_rules! set_table_vnode_mappings {
```

Comment:
`set_table_vnode_mappings` are used in both `Scheduler` and `FragmentManager`, which might lead to identical codes (function definition) in different modules. So to make code cleaner, I just use macro.

Reply:
Yes, I understand the code need to be resued, but why not do it just with a function? So you call it from 2 places

Reply:
I found nowhere to place this function... It's weird not to be defined as a method of a struct, and defining it in either `Scheduler` or `FragmentManager ` is undesirable.

Reply:
Either

1. Place it as a normal function (i.e. just replace `&self` to `hash_mapping_manager: &HashMappingManagerRef`)
2. Place it as a function of `HashMappingManager` 

looks okay to me, I think it will be better than macro

Reply:
Macro is simple text replacement. Therefore, it is out of the control of any coding convention or compiler checks, which makes other developers harder to understand your code.

For example, in this case, another develop must read the implementation to get to know:

- the macro must be used in a struct
- `self.hash_mapping_manager` must be in struct
- `self.hash_mapping_manager` must have a function `set_fragment_state_table`. So what's the type of it???

On the other side, if another develop or yourself wants to modify this code, the compiler can only check the code after macro expanded. As a result, the compiler error is much more harder to read.

Reply:
I'm convinced. I'll use function instead.